src/lib/Util.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity ^0.8.0;
+
+library Util {
+    error NegativeUint();
+    error IndexOutOfBounds();
+    error InvalidArrLength();
+
+    function subIntFromUint(int256 a, uint256 b) public pure returns (uint256) {
+        if (a < 0) {
+            // If a is negative, add its absolute value to b
+            return b + uint256(-a);
+        } else {
+            // If a is positive, subtract it from b
+            if (b < uint256(a)) {
+                revert NegativeUint();
+            }
+
+            return b - uint256(a);
+        }
+    }
+
+    // Retrieve the nth uint16 value from a packed uint256
+    function getUint16FromPacked(uint256 packedData, uint256 n) public pure returns (uint16) {
```

Comment:
Consider using a custom type a la https://github.com/Uniswap/v4-core/blob/main/src/types/BalanceDelta.sol

to represent 16xuint16s

Reply:
Defining a custom type for the test case `testGetUint16FromPacked` increases gas cost by 356. Do you think the readibility is worth the gas trade-off?